pkg/handlers/webhookrhobsreceiver.go

```diff
@@ -0,0 +1,222 @@
+	ocm OCMClient
+}
+
+func NewWebhookRHOBSReceiverHandler(c client.Client, o OCMClient) *WebhookRHOBSReceiverHandler {
+	return &WebhookRHOBSReceiverHandler{
+		c:   c,
+		ocm: o,
+	}
+}
+
+func (h *WebhookRHOBSReceiverHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// validate request
+	if r != nil && r.Method != http.MethodPost {
+		http.Error(w, "Method Not Allowed", http.StatusMethodNotAllowed)
+		return
+	}
+	var err error
+	var alertData AMReceiverData
+	err = json.NewDecoder(r.Body).Decode(&alertData)
+	if err != nil {
+		log.Errorf("Failed to process request body: %s\n", err)
+		http.Error(w, "Bad request body", http.StatusBadRequest)
+		metrics.SetRequestMetricFailure(consts.WebhookReceiverPath)
+		return
+	}
+
+	// process request
+	response := h.processAMReceiver(alertData, r.Context())
+
+	// write response
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(response.Code)
+	err = json.NewEncoder(w).Encode(response)
+	if err != nil {
+		log.Errorf("Failed to write to response: %s\n", err)
+		http.Error(w, "Failed to write to response", http.StatusInternalServerError)
+		metrics.SetRequestMetricFailure(consts.WebhookReceiverPath)
+		return
+	}
+
+	metrics.ResetMetric(metrics.MetricRequestFailure)
+}
+
+func (h *WebhookRHOBSReceiverHandler) processAMReceiver(d AMReceiverData, ctx context.Context) *AMReceiverResponse {
+	log.WithField("AMReceiverData", fmt.Sprintf("%+v", d)).Info("Process alert data")
+
+	// Handle each firing alert
+	for _, alert := range d.Alerts.Firing() {
+		// Can we find a notification template for this alert?
+		templateName := alert.Labels[AMLabelTemplateName]
+		mfn := oav1alpha1.ManagedFleetNotification{}
+		err := h.c.Get(ctx, client.ObjectKey{
+			Namespace: OCMAgentNamespaceName,
+			Name:      templateName,
+		}, &mfn)
+		if err != nil {
+			log.WithError(err).Error("unable to locate corresponding notification template")
+			return &AMReceiverResponse{Error: err,
+				Status: fmt.Sprintf("unable to find ManagedFleetNotification %s", templateName),
+				Code:   http.StatusInternalServerError}
+		}
+
+		// Filter actionable alert based on Label
+		if !isValidAlert(alert, true) {
+			log.WithField(LogFieldAlert, fmt.Sprintf("%+v", alert)).Info("alert does not meet valid criteria")
+			continue
+		}
+
+		err = h.processAlert(alert, mfn)
+		if err != nil {
+			log.WithError(err).Error("a firing alert could not be successfully processed")
+		}
+	}
+	return &AMReceiverResponse{Error: nil, Status: "ok", Code: http.StatusOK}
+}
+
+// processAlert handles the pre-check verification and sending of a notification for a particular alert
+// and returns an error if that process completed successfully or false otherwise
+func (h *WebhookRHOBSReceiverHandler) processAlert(alert template.Alert, mfn oav1alpha1.ManagedFleetNotification) error {
+	fn := mfn.Spec.FleetNotification
+	mcID := alert.Labels[AMLabelAlertMCID]
+	hcID := alert.Labels[AMLabelAlertHCID]
+	slSent := false
+	for !slSent {
```

Comment:
why do we need a for loop here?

Reply:
Yeah I don't think a loop is needed at all, we can get rid of that.

Reply:
This has now been addressed (for loop removed)